### PR TITLE
New module: coin_balance

### PR DESCRIPTION
**Pull request by fmorgner**
The 'coin_balance' module fetches the users balances of different
crypto-currencies from a user-defined server. The currently supported
coins are: Bitcoin, Dogecoin, and Litecoin. 'coin_balance' requires a
server that is compatible with the Bitcoin RPC protocol and is easily
extendable to new crypto-currencies by extending the 'COIN_PORTS'
dictionary.

### CODE REVIEW

File: py3status/modules/coin_balance.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            with open(expanduser('~/.{0}/{0}.conf').format(coin)) as cfg:
+                creds = {}
+                for line in cfg.readlines():
+                    fields = line.split('=')
+                    if len(fields) == 2 and fields[0] in CREDENTIAL_FIELDS:
+                        creds[CREDENTIAL_FIELDS[fields[0]]] = fields[1].strip()
+
+                if all(key in creds for key in CREDENTIAL_FIELDS.values()):
+                    return creds
+        except FileNotFoundError:
```
Comment (tobes) [inline py3status/modules/coin_balance.py]: This exception  does not exist in python 2.x

File: py3status/modules/coin_balance.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            pass
+
+            return None
```
Comment (tobes) [inline py3status/modules/coin_balance.py]: This return is unneeded as the default action of a function is to return None

**Comment by tobes** [comment]
Does the user need a way to set their credentials?

**Comment by fmorgner** [comment]
@tobes Thanks for the feedback! Since I'am somewhat new to python, would you recommend just capturing IOError instead of FileNotFoundError? Regarding credentials, i will implement that ASAP.

**Comment by tobes** [comment]
`OSError` catches a wider bunch but probably fills your need here.

```
from errno import ENOENT
...
except OSError as e:
    if e.errno == ENOENT:
        # file not found
```
is the equivalent of  FileNotFoundError that works in python 2/3 but is horrible

**Comment by pydsigner** [comment]
IOError would be better than OSError if it works.

**Comment by ultrabug** [comment]
Thanks a lot for your proposal @fmorgner !

I'll let you take @tobes and @pydsigner remarks into account, they're great reviewers and this shall be a great way to improve while making a better module :)

File: py3status/modules/coin_balance.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            }
+
+        return self._credential_cache[coin]
+
+    def _get_balance(self, coin):
+        if coin not in COIN_PORTS:
+            return 'Unsupported coin'
+
+        try:
+            credentials = self._get_credentials(coin)
```
Comment (tobes) [inline py3status/modules/coin_balance.py]: This feels like it should be before the `try`

**Comment by tobes** [comment]
@ultrabug apart from the last comment I've just added the others have been done.  I can't test this but it looks good to me.  There is a raw except in `_get_balance()` that I don't like but I think here I can live with it.

**Comment by ultrabug** [comment]
Thanks a lot @fmorgner !

**Comment by fmorgner** [comment]
Thanks for the support from all of you!

**Pull request merged**